asv/commands/publish.py

```diff
@@ -177,8 +177,12 @@ def run(cls, conf):
         for key in six.iterkeys(benchmark_map):
             check_benchmark_params(key, benchmark_map[key])
         for key, val in six.iteritems(params):
-            val = list(val)
-            val.sort(key=lambda x: x or '')
+            # Values can be stored as strings, integers, floating point values,
+            # or None, so we now convert to a list of strings. We also make 
+            # sure that we convert to a set to remove duplicate items (for 
+            # instance, some files might contain "" and some None, but we need 
+            # to treat them as one).
+            val = sorted(set(str(x) if x is not None else '' for x in val))
```

Comment:
Indeed - and I think back-compat is important in this case since we need to be able to include old results in the graphs.